CMakeLists.txt

```diff
@@ -28,14 +40,30 @@
 #                        ISOLATION_LEVEL 2
 #                        REGRESSION
 #                        BL2 True
-#                        BUILD_PROFILE profile_small)
+#                        BUILD_PROFILE profile_small
+#                        TFM_PARTITION_PROTECTED_STORAGE ON
+#                        TFM_PARTITION_INTERNAL_TRUSTE_STORAGE ON
+#                        TFM_PARTITION_CRYPTO ON
+#                        TFM_PARTITION_INITIAL_ATTESTATION ON
+#                        TFM_PARTITION_PLATFORM ON
+#                        TFM_PARTITION_AUDIT_LOG ON)
 function(trusted_firmware_build)
   set(options IPC REGRESSION)
-  set(oneValueArgs BINARY_DIR BOARD BL2 ISOLATION_LEVEL CMAKE_BUILD_TYPE BUILD_PROFILE)
+  set(oneValueArgs BINARY_DIR BOARD BL2 ISOLATION_LEVEL CMAKE_BUILD_TYPE
+    BUILD_PROFILE ${tfm_partitions})
   cmake_parse_arguments(TFM "${options}" "${oneValueArgs}" "" ${ARGN})
 
-  if(NOT DEFINED TFM_BL2)
-    set(TFM_BL2 True)
+  # Set all arguments which enable/disable the secure partitions.
+  foreach(partition ${tfm_partitions})
+    set(val ${TFM_${partition}})
```

Comment:
This means that using this function, it is now a requirement to specify ALL partitions.
Is that really needed ?
Why not default each partition to `OFF`, and then use the argument name: `PARTITIONS` in a multi args argument, see other comment.

Reply:
That sounds much better.